.github/actions/build-avr/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM alpine
+
+RUN apk add --update --no-cache curl gcompat
+# libc6-compat advice from here https://githubmemory.com/repo/arduino/arduino-cli/issues/1378
+# but that doesn't work and gives the error described at https://www.gitmemory.com/issue/arduino/Arduino/5691/757103707
+# using gcompat makes the cli and the compiler work
+
+ENV ARDUINO=/Arduino
+ENV ARDUINO_CLI=/Arduino/cli
+ENV ARDUINO_DIRECTORIES_USER=$ARDUINO
+ENV WORKSPACE=${GITHUB_WORKSPACE}
+ENV PATH="${ARDUINO_CLI}/bin:${PATH}"
+
+RUN curl -fsSL https://raw.githubusercontent.com/arduino/arduino-cli/master/install.sh | sh && \
+    mkdir -p $ARDUINO_CLI/bin && \
+    mkdir -p $ARDUINO/libraries && \
+    arduino-cli config init && \
+    arduino-cli core update-index && \
+    arduino-cli core install arduino:avr && \
+    arduino-cli core install arduino:avr && \
+    arduino-cli core install arduino:samd
```

Comment:
It depends on what boards (and thus architectures) we want to target.

Reply:
You have specifically created the `build-avr` build action. I would expect that we have an analogous `build-samd` build action to build the examples for that architecture. I like the idea of single purpose build actions.

Reply:
I see what you mean. When I started it, I was just thinking avr, since I didn't know we targeted other architectures at the time.  I'll give the base build a more suitable name. We can certainly split out more complex architectures, but I would only do that when the overhead is worth it, as is the case with esp32/nrf52 that require an ubuntu-based build and various dependencies. 

With multiple architectures in different action directories, one concern is reusing the common elements rather than copying scripts across directories, but I'm pretty sure that's easy to arrange.